### PR TITLE
ICU-21569 Fix Java7 breakage in LSTM & add Java7 CI

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21569
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [X] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: .github/workflows/icu_ci.yml [hunk 1]
```diff
@@ -30,6 +30,47 @@ jobs:
...
+      - name: Checkout and setup
+        uses: actions/checkout@v2
+        with:
+          lfs: true
+      - name: Checkout lfs objects
+        run: git lfs pull
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '8'
+      - name: Cache for Java 7 tarball
```
Comment (echeran) [inline .github/workflows/icu_ci.yml:46]: Adding this caching step, as you've done here, and using it in the later steps, should work. I've tested that the [first run](https://github.com/echeran/icu/pull/4/checks?check_run_id=2689066843#step:5:9) will get a cache miss and the [second run](https://github.com/echeran/icu/runs/2689212122?check_suite_focus=true#step:5:14) will get a cache hit. That insulates will insulate us from the flakiness of network connections and failed downloads.

**Review by echeran — APPROVED** [review]
LGTM

**Review by echeran — APPROVED** [review]
rubber stamp after the squash. I also saw Frank verify the cache miss (1st run) & cache hit behavior (2nd run), so LGTM.

**Review by markusicu — APPROVED** [review]
lgtm tnx!

**Pull request merged**